tests/data/greek-verb-paradigm.test.js

```diff
@@ -0,0 +1,856 @@
+/* eslint-env jest */
+/* eslint-disable no-unused-vars */
+import 'whatwg-fetch'
+import { ClientAdapters } from 'alpheios-client-adapters'
+import { Constants, Feature, LanguageModelFactory } from 'alpheios-data-models'
+
+import ViewSetFactory from '@views/lib/view-set-factory.js'
+// import LanguageDatasetFactory from '@views/lib/language-dataset-factory.js'
+
+describe('greek-verb-paradigm.test.js', () => {
+  console.error = function () {}
+  console.log = function () {}
+  console.warn = function () {}
+
+  const locale = "en-US"
+  beforeAll(async () => {
+  })
+
+  beforeEach(() => {
+    jest.spyOn(console, 'error')
+    jest.spyOn(console, 'log')
+    jest.spyOn(console, 'warn')
+  })
+  afterEach(() => {
+    jest.resetModules()
+  })
+  afterAll(() => {
+    jest.clearAllMocks()
+  })
+
+  async function getInflectionSet(targetWord, languageID) {
+    const adapterTuftsRes = await ClientAdapters.morphology.tufts({

```

Comment:
this is okay for now, but ultimately I would like to switch to using fixtures for the Homonyms so that we don't have service dependencies in our unit tests.

Reply:
I thought about it but it is not really a simple task.
The Client-Adapters repo than should be changed, because for now

Client-Adapters has three functions - prepare input data, retrieve result from the service and convert data.
We need to stub only retrieve. So I believe I need to create a state of ClientAdapters when it doesn't retrieve data, but gets it from input parameters.
May be itis time to work on it too.

Reply:
Let's create a issue for it.

Reply:
yes, I agree with @kirlat 's suggestion.  @irina060981 we may also need a change such as you suggest to Client adapters for integration testing, so I think I agree with both of you :-)

Reply:
>What if we create a stub function just for testing that
I don't agree with this because:
  - these are not unit tests - these tests are integration tests by its sense, so using the real library that returns words are justified
  - I used a lot of words for tests and may be in future there will be more words.
    If I will need to create a stub function for it of them then I would need to create a lot of Homonyms with all its inner Lexemes, Lemmas, Inflections and so on manually. And for new one I (or someone else) would need to create it again and again. And it would be really difficult to create and be sure that such stubs are correct. 
    If I would go with Json converter similiar that WordList Controller do - that I need to create a separate methods to imitiate creation Homonyms that would not be really similiar to those we have. And would need its own verification.

So I believe that it is too much manual work and finally We couldn't be sure that it would return correct, consistent and useful data for checking inflections.

Also we use Client-Adapters in much tests (you could see it if you read tests in components and other repos) and it would need a lot of duplicated stub job.

That's why I suggest 
    - upload recieved data for different homonyms and morph services in the specific folder
    - create an alternative method of get the data - that would need only one small change - get data not from the remote source but from the local file

This way it could be easily created, easily checked, easily extended by other words and could be used for several adapters (not only morph).

Reply:
hmm. so you are suggesting essentially that the mock data go into the client-adapters repository rather than that of the client-adapter clients. I think I didn't fully understand that before. I think that's actually a great idea, and is in some senses not too different from the idea of a stub method, just implemented somewhat differently. 

The one concern I do have with this though is that it results in alot of test data going in the built package (and needing to be downloaded by the user at runtime). I think we would need a way to exclude that data from the runtime built package. @kirlat what do you think?

Reply:
I think it's very good idea to move this testing code along with related data to ClientAdapters because it, ultimately, belongs to ClientAdapters much more than to inflection tables. This testing code an be reused by other libraries too. 

I, however, agree with @balmas that it shall not be mixed with production code of ClientAdapters. I think there can be several approaches in solving this. One way is to create a test version of "get data" method and place it along with test data into the test directory of ClientAdapters. A a test suite of a client library (inflection tables, for example) can import this method directly from node-modules/alpheios-client-adapters/test and replace a production method with a test one. This way there will be no test code in business logic of ClientAdapters. 

What do you think about that?

Reply:
may be it is better to go another way - 
I would create another repo for test fixtures
There I would use client-adapters repo and create a children Client-Adapter class,
that would use test data instead of real source service data.

So it would use the same format of methods
     it would import classes naturally without direct access to node-modules
     it could be imorted as dev-dependencies and won't be included in production bundle (I believe)

@kirlat and @balmas , what do you think?

Reply:
I think it is a great idea but, on my opinion, creating a separate repository just for a test version of client adapters would be an overkill. What if to create a separate directory in client adapters for test code and test data? Since it's separate and nothing from it is mentioned in `index.js` it won't be included into a main client adapters bundle. Then we can create a separate `index.js` inside a test-related directory and a separate build configuration in `build`. That test build configuration can take `index.js` from a test directory as source and create a different bundle within the `dist`. That bundle can then be used for testing purposes when needed. What do you think abut such approach?

Reply:
Maybe it's just work avoidance, but after reading this https://medium.com/@copyconstruct/testing-microservices-the-sane-way-9bb31d158c16 (referenced from the article @kirlat shared) particularly the bit about the "The Step Up Rule" I am wondering if maybe my bias against use of the live morphology service in unit tests is unwarranted.

Reply:
I think that in our case it is worth to do in the way I suggested, when we stub only the service result and nothing more.
So we simply will take off the load from morphology, definitions, concordance services and others. Now our tests could have much load for the services at USA working hours.
But all other aspects will be the same - the same code, the same workflow and if we will choose the proper way to do that - it won't take even much time for developing such fixture and would be easily extendable.
That's why i suggest to realize it in the separate repo - this way we would be able to add new stubs for work and services without updating the master Client-Adapters repo.
It is from my point of view.